### PR TITLE
Refactor/methods

### LINKED ISSUE #45: User/recipe methods
**Description of what needs refactoring**
Move qualifying functions to be methods on the user and recipe classes
Update test files for each class method

**Additional Context** 
Should be able to reduce scripts length by moving additional functions into separate files if possible

### LINKED ISSUE #50: Dataset References
**Description of what needs refactoring**
All uses of datasets from data source folder should be removed and used only for testing 

**Additional Context** 
Files can be updated to include reduced datasets for testing purposes

### LINKED ISSUE #72: Dry up finding recipes
**Description of what needs refactoring**
A brief description of the code that needs refactoring and the reason for refactoring
- this line of code appears like 5 times in scripts 
`allRecipes.find(recipe => recipe.id === Number(recipeId))` 

- we could make a function for that 

**Additional Context** 
File/function name or additional reasoning for the changes

**Screenshot of Issue**
Include a screenshot if it would help describe the issue

### LINKED ISSUE #73: Dry up finding recipe
**Description of what needs refactoring**
A brief description of the code that needs refactoring and the reason for refactoring
- this line of code appears like 5 times in scripts 
`allRecipes.find(recipe => recipe.id === Number(recipeId))` 

- we could make a function for that 

**Additional Context** 
File/function name or additional reasoning for the changes

**Screenshot of Issue**
Include a screenshot if it would help describe the issue

### PR DESCRIPTION
**Pull request by richardltyler**
## *Purpose:*
*What does this merge do? Check all that apply.*
- [ ] Adds a new feature
- [x] Improves an existing feature
- [ ] Fixes a bug
- [x] Cleans up (logs, tests, etc.)

## *Description: What does this PR do?*
*Briefly, but clearly describe what you did, and explain any new code.*
- moves refactored applicable functions to be methods on user and recipe classes
- fixed data in repo 

## *Instructions for reviewer:*
*Where should the reviewer start?*
- `user.js`
*How should this be manually tested?*
- run `npm test` in command line 
*Any background context you want to provide?*
*What are the relevant tickets?*
closes #73 
closes #72 
closes #50
closes #45

## *Issues:*
*Are there any concerns, issues, or bugs in this branch? If so describe them.*

## Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have used tests to prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my change

### CODE REVIEW

File: src/scripts.js [hunk 1]
```diff
@@ -55,13 +52,13 @@ Promise.all([fetchedUserData, fetchedRecipeData, fetchedIngredientData])
...
     addRecipeNameAndCost(values[2])
-    addPantryIngredientNames(values[2])
+    // currentUser.addPantryIngredientNames(values[2])
     loadWebsite()
   })
 
 function loadWebsite() {
   createRecipeCards()
-  displayPantryInfo(currentUser)
+  domUpdates.addPantryInfoToDom(currentUser.alphabetizePantry())
```
Comment (richardltyler) [inline src/scripts.js]: I'm not totally sure about this syntax. I would be open to suggestions about readability or something

File: src/scripts.js [hunk 2]
```diff
@@ -55,13 +52,13 @@ Promise.all([fetchedUserData, fetchedRecipeData, fetchedIngredientData])
     generateUser(values[0])
     createRecipeDataset(values[1])
     addRecipeNameAndCost(values[2])
-    addPantryIngredientNames(values[2])
+    // currentUser.addPantryIngredientNames(values[2])
```
Comment (alia-peterson) [inline src/scripts.js]: Looks like this line can be removed since it's currently commented out and being called elsewhere

File: src/scripts.js [hunk 3]
```diff
@@ -81,37 +78,22 @@ function createRecipeDataset(recipeInfo) {
...
 
-// USER PANTRY DISPLAY AND UPDATES
-function displayPantryInfo(currentUser) {
-  currentUser.pantry.sort(function(a, b) {
-    if (a.name > b.name) {
-      return 1
+// FIND A SPECIFIC RECIPE
+function findRecipe(recipeId) {
+  return allRecipes.find(recipe => recipe.id === Number(recipeId))
+}
```
Comment (alia-peterson) [inline src/scripts.js:96]: nice! such a useful refactor

File: src/scripts.js [hunk 4]
```diff
@@ -145,22 +127,16 @@ async function updateUserPantryDisplay(recipeId, typeModification = 'add') {
   await updateUserPantryFromRecipe(recipeId, typeModification)
   await retrieveUpdatedUserPantry()
   await fetchedIngredientData.then(array => {
-    addPantryIngredientNames(array)
+    currentUser.addPantryIngredientNames(array)
```
Comment (alia-peterson) [inline src/scripts.js]: i named this variable array but i feel like we should rename it now that we've figured out the functionality. "Ingredients" maybe.

File: src/user.js [hunk 5]
```diff
@@ -4,7 +4,7 @@ class User {
     this.name = user.name
     this.pantry = user.pantry
     this.favoriteRecipes = []
-    this.recipesToCook = []
+    this.cookedRecipes  = []
```
Comment (alia-peterson) [inline src/user.js]: Nit-picky buuuut there's an extra space before the equal sign

File: src/user.js [hunk 6]
```diff
@@ -20,20 +20,24 @@ class User {
...
+    return this.pantry.sort(function(a, b) {
+      if (a.name > b.name) {
+        return 1
+  
+      } else if (a.name < b.name) {
+        return -1
+      }
     })
   }
+
```
Comment (alia-peterson) [inline src/user.js]: Also nit-picky but I don't think this line-break is useful since it's the end of the class.

**Review by alia-peterson — APPROVED** [review]
- [ ] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] No spelling errors
- [x] Approved changes with team
- [x] All teammates understand changes

File: src/scripts.js (hunk 1 above)
Comment (richardltyler) [reply src/scripts.js]: I went ahead and assigned that user method to a variable then passed it in for readability

Comment (tashiad) [reply src/scripts.js]: looks ok to me... pretty sure we're doing a similar thing in other places

File: src/scripts.js [hunk 7]
```diff
@@ -81,37 +78,22 @@ function createRecipeDataset(recipeInfo) {
...
+  currentUser.addPantryIngredientNames(allIngredients)
 }
 
-// USER PANTRY DISPLAY AND UPDATES
-function displayPantryInfo(currentUser) {
-  currentUser.pantry.sort(function(a, b) {
-    if (a.name > b.name) {
-      return 1
+// FIND A SPECIFIC RECIPE
+function findRecipe(recipeId) {
```
Comment (tashiad) [inline src/scripts.js:94]: yaaaasss

File: test/recipe-test.js [hunk 8]
```diff
@@ -3,13 +3,21 @@ import { expect } from 'chai'
 import Recipe from '../src/recipe'
 import data from '../src/data/recipe-data'
 
+import ingredientsData from '../src/data/ingredient-data'
+
 describe('Recipe', function() {
   let recipe
   let recipeInfo
+  let recipe2Info
```
Comment (tashiad) [inline test/recipe-test.js:11]: Nitpicky comment but I'd swap lines 11 and 12 to be consistent with the order you're using for recipe & recipeInfo

**Review by tashiad — APPROVED** [review]

**Review by alia-peterson — APPROVED** [review]
There was one merge conflict. I'm pretty sure I merged them correctly but we'll want to make sure nothing broke. 

- [ ] Follows Turing Style Guidelines
- [x] Changes generate no new warnings or errors
- [x] No spelling errors
- [x] Approved changes with team
- [x] All teammates understand changes

**Pull request merged**